std/range/package.d

```diff
@@ -7125,9 +7131,64 @@ private:
     size_t _chunkSize;
 }
 
+/// ditto
+struct Chunks(Source)
+if (isInputRange!Source && !isForwardRange!Source)
+{
+    static struct Chunk
+    {
+        private Impl* impl;
+
+        @property bool empty() { return impl.curSizeLeft == 0 || impl.r.empty; }
+        @property auto front() { return impl.r.front; }
+        void popFront()
+        {
+            assert(impl.curSizeLeft > 0 && !impl.r.empty);
+            impl.curSizeLeft--;
+            impl.r.popFront();
+        }
+    }
+
+    static struct Impl
+    {
+        private Source r;
+        private size_t chunkSize;
+        private size_t curSizeLeft;
+        private bool _empty;
+    }
+
+    private Impl* impl;
+    alias impl this;
+
+    private this(Source r, size_t chunkSize)
+    {
+        impl = new Impl(r, chunkSize, chunkSize);
```

Comment:
Two more arguments for `RefCounted`:
- making safe RC (`core.rc`) is [on the agenda](https://wiki.dlang.org/Vision/2017H2_draft), so if we use `RefCounted` now, chances are good that it might turn `@safe` in the future
- it's easier to use `@trusted` on your code than the hacky and rather unknown [`assumeNoGc`](https://github.com/nordlow/phobos-next/blob/200becc58d9e37cd7866d01c017a3dfcbed84de9/src/dbgio.d#L13).
In any case it's probably a good idea to open an issue about it, s.t. it's not forgotten.